### PR TITLE
Qute: introduce InjectableFragment

### LINKED ISSUE #165: Wrong Content-Type header when using fragments
When the response is an HTML fragment it adds the `Content-Type:text/plain;` instead of  `Content-Type:text/html;` so the page is not rendered properly.  

```java
@Path("/")
public class HelloResource extends io.quarkiverse.renarde.Controller {

    public io.quarkus.qute.TemplateInstance index() {
        return HtmlTemplate.index();
    }

    public io.quarkus.qute.TemplateInstance fragment() {
        return HtmlTemplate.index$fragment();
    }

}
```

```html
<p style="color:red">from qute index</p>

{#fragment id='fragment' rendered=false}
 <p style="color:red">the fragment</p>
{/fragment}
```

```java
@CheckedTemplate
public class HtmlTemplate {

    public static native TemplateInstance index();

    public static native TemplateInstance index$fragment();

}
```

### PR DESCRIPTION
**Pull request by mkouba**
- type-safe fragments should honor the selected variant
- fixes https://github.com/quarkiverse/quarkus-renarde/issues/165

### CODE REVIEW

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/TemplateProducer.java [hunk 1]
```diff
@@ -280,6 +341,22 @@ private Template template() {
             }
 
         }
+
+        class InjectableFragmentTemplateInstnaceImpl extends InjectableTemplateInstanceImpl {
```
Comment (FroMage) [inline extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/TemplateProducer.java:345]:
```suggestion
        class InjectableFragmentTemplateInstanceImpl extends InjectableTemplateInstanceImpl {
```

Comment (FroMage) [reply extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/TemplateProducer.java:345]: Typo :)

**Review by FroMage — APPROVED** [review]
I marked it as approve just to avoid having to re-review for a typo, but look great ;)

File: extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/TemplateProducer.java (hunk 1 above)
Comment (mkouba) [reply extensions/qute/runtime/src/main/java/io/quarkus/qute/runtime/TemplateProducer.java:345]: 🤦‍♂️ thanks...

**Pull request merged**